src/servicesHost.ts

```diff
@@ -505,17 +514,9 @@ function resolveModuleName(
     // tslint:disable-next-line:no-empty
   } catch (e) {}
 
-  const tsResolver = (moduleName: string, containingFile: string, compilerOptions: typescript.CompilerOptions, moduleResolutionHost: typescript.ModuleResolutionHost) =>
-    compiler.resolveModuleName(
-      moduleName,
-      containingFile,
-      compilerOptions,
-      moduleResolutionHost
-    );
-
   const tsResolution = customResolveModuleName
-    ? customResolveModuleName(moduleName, containingFile, compilerOptions, moduleResolutionHost, tsResolver)
-    : tsResolver(moduleName, containingFile, compilerOptions, moduleResolutionHost);
+    ? customResolveModuleName(moduleName, containingFile, compilerOptions, moduleResolutionHost, applyTsResolver.bind(null, compiler))
```

Comment:
I hadn't expected the `.bind(null, compiler)` would be necessary... I'm guessing it is?  What happens without it? Presumably some kind of `this` issue?

Reply:
No, we simply need the `compiler` variable to be able to call `compiler.resolveModuleName` (and `compiler` is obtained from `instance`, which isn't a global)